### PR TITLE
Add url and typing hint for BFS

### PR DESCRIPTION
**Pull request by wuyudi**
### **Describe your change:**

Add typing for bfs and make the code more brief.

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: graphs/bfs.py [hunk 1]
```diff
@@ -26,13 +28,12 @@
...
 
-def bfs(graph, start):
+def bfs(graph: Dict, start: str) -> Set[int]:
     """
     >>> ''.join(sorted(bfs(G, 'A')))
     'ABCDEF'
     """
-    explored, queue = set(), [start]  # collections.deque([start])
-    explored.add(start)
+    explored, queue = set(start), [start]  # collections.deque([start])
```
Comment (cclauss) [inline graphs/bfs.py:36]: `explored` would be different if `start` was more than a single character.

File: graphs/bfs.py [hunk 2]
```diff
@@ -26,13 +28,12 @@
 }
 
 
-def bfs(graph, start):
+def bfs(graph: Dict, start: str) -> Set[int]:
```
Comment (cclauss) [inline graphs/bfs.py:31]:
```suggestion
def breadth_first_search(graph: Dict, start: str) -> Set[int]:
```

File: graphs/bfs.py (hunk 1 above)
Comment (cclauss) [inline graphs/bfs.py:36]:
```suggestion
    explored = {start}
    queue = [start]
```

Comment (wuyudi) [reply graphs/bfs.py]: So... how to fix it?

**Comment by cclauss** [comment]
You will need to change all instances of bfs() to breadth_first_search().

**Comment by cclauss** [comment]
Do not worry about Euler 47

File: graphs/bfs.py [hunk 3]
```diff
@@ -1,8 +1,6 @@
-"""
+"""https://en.wikipedia.org/wiki/Breadth-first_search
 BFS.
```
Comment (cclauss) [inline graphs/bfs.py:2]:
```suggestion
"""
https://en.wikipedia.org/wiki/Breadth-first_search
```

File: graphs/bfs.py [hunk 4]
```diff
@@ -1,8 +1,6 @@
...
-
 pseudo-code:
-
 BFS(graph G, start vertex s):
```
Comment (cclauss) [inline graphs/bfs.py:4]:
```suggestion
breadth_first_search(graph G, start vertex s):
```

**Comment by cclauss** [comment]
We already have a breadth_first_search.py.  Should this file be deleted??

**Comment by wuyudi** [comment]
There are two different dfs files.
https://github.com/TheAlgorithms/Python/blob/master/graphs/depth_first_search.py
https://github.com/TheAlgorithms/Python/blob/master/graphs/depth_first_search_2.py
The same as these two bfs files.
https://github.com/TheAlgorithms/Python/blob/master/graphs/bfs.py
https://github.com/TheAlgorithms/Python/blob/master/graphs/breadth_first_search.py

**Pull request merged**